### PR TITLE
Remove TokenId Validation

### PR DESCRIPTION
**Pull request by 0xSpraggins**
# Description

Currently wrapper token Ids are used to store the starting tokenPerShare value for rewards. This is not entirely necessary to be stored within an id but it is fine as this codebase has already been through countless audits with this format and should not be changed. However, it is unnecessary for Id's to be unique as accounting is done within the Bank through `pos.collateralSize`.  This can cause issues further down the line if conflicting tokenIds are created the contract can be bricked if multiple positions are opened within the same block or before tokenPerShare has been updated. In order to avoid this we should remove the check `_validateTokenId` from all wrappers. This PR accomplished this while also deleting the `BaseWrapper` contract which only holds this internal function.

## Type of change

- [x] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests are included for all code paths
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: contracts/wrapper/WAuraBooster.sol [hunk 1]
```diff
@@ -163,8 +167,6 @@ contract WAuraBooster is IWAuraBooster, BaseWrapper, ReentrancyGuardUpgradeable,
         uint256 balRewardPerToken = IRewarder(auraRewarder).rewardPerToken();
         id = encodeId(pid, balRewardPerToken);
 
-        _validateTokenId(id);
```
Comment (mayorcoded) [inline contracts/wrapper/WAuraBooster.sol:166]: My observation here is that the validation prevents the sender from minting more units of a previously minted token, are we okay with allowing that happen now?

**Review by mayorcoded — COMMENTED** [review]
Looks good overall, just left a question.

File: contracts/wrapper/WAuraBooster.sol (hunk 1 above)
Comment (0xSpraggins) [reply contracts/wrapper/WAuraBooster.sol:166]: Your observation is correct but this is the exact reason we need to remove this validation. The whole reasoning for tokenId's is to store the starting point of `rewardPerToken` for each position. If two users happen to enter the position immediately after one another there is a chance that `rewardPerToken` will be the same. All accounting is done within the bank stating how many of the wrapper tokens a user has, which ultimately determines all reward calculations within the contract.

Comment (mayorcoded) [reply contracts/wrapper/WAuraBooster.sol:166]: Ah, I get it now, thanks for clarifying.

**Review by mayorcoded — APPROVED** [review]

**Review by Gornutz — APPROVED** [review]
- [x] Peer-reviewed

**Review by cuthalion0x — APPROVED** [review]

**Pull request merged**